### PR TITLE
wip: sample db

### PR DESCRIPTION
**Pull request by yehjxraymond**

### CODE REVIEW

**Comment by sebastianquek** [comment]
Some thoughts about using context here. 2 questions come to mind:
1. How do we initialise the store/db for consumption by all nested children?
2. How do we talk to the store when accessing it using context?

For 1, the current approach uses `useReducer` to expose the `dispatch` method so that we can dispatch the `SET_DB` action. However, for 2, instead of using `dispatch` to talk to the store, the `db` is directly used. This poses the question of whether `dispatch` should be used.

I suggest we stick to not using `dispatch` to reduce the amount of boilerplate needed (e.g. actions, reducers, etc.). To handle 1, we can do the following. Let me know what you think!
```
import React, { SetStateAction, Dispatch } from 'react';
import { RxDatabase } from "rxdb";

export const StoreContext= React.createContext<{
  store: RxDatabase | null;
  setStore: Dispatch<SetStateAction<RxDatabase>>;
}>({
  store: null,
  setStore: () => null,
});
```

```
const App = () => {
  const [store, setStore] = useState(null);

  return (
    <StoreContext.Provider value={{store, setStore}}>
      <AppContainer />
    </StoreContext.Provider>
  );
};
```

```
const {setStore} = useContext(StoreContext);
useEffect(() => {
  createDatabase().then(db => {
    setStore(db);
    navigation.navigate("StackNavigator");
  });
}, [])
```

File: src/context/db.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export const DbContext = createContext({});
+
+export const useDbContext = (): { db: RxDatabase; setDb: Function } =>
+  useContext(DbContext) as any;
+
+interface DbContextProviderProps {
+  children: object;
+}
+
+export const DbContextProvider = ({ children }: DbContextProviderProps) => {
```
Comment (sebastianquek) [inline src/context/db.tsx]: We can do `export const DbContextProvider: React.FC = ({ children }) => {`, which will automatically include the `children` prop

File: src/context/db.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React, { createContext, useContext, useState } from "react";
+import { RxDatabase } from "rxdb";
+
+export const DbContext = createContext({});
+
+export const useDbContext = (): { db: RxDatabase; setDb: Function } =>
+  useContext(DbContext) as any;
```
Comment (sebastianquek) [inline src/context/db.tsx]: I noticed the `any` here, how strict do we intend to be with this 😊

Comment (yehjxraymond) [reply src/context/db.tsx]: fixed

File: src/context/db.tsx [hunk 3]
```diff
@@ -1,16 +1,30 @@
...
+  db: undefined,
+  setDb: undefined
+});
 
-export const useDbContext = (): { db: RxDatabase; setDb: Function } =>
-  useContext(DbContext) as any;
+export const useDbContext = (): DbContext => useContext<DbContext>(DbContext);
 
-interface DbContextProviderProps {
+export interface DbContextProviderProps {
```
Comment (sebastianquek) [inline src/context/db.tsx]: Hmm do we still need this since we only use `children`?

File: src/context/db.tsx (hunk 1 above)
Comment (yehjxraymond) [reply src/context/db.tsx]: With React.FC it's still showing `  16:47  error  'children' is missing in props validation  react/prop-types`

File: src/context/db.tsx (hunk 3 above)
Comment (yehjxraymond) [reply src/context/db.tsx]: seemed like so

**Comment by yehjxraymond** [comment]
@sebastianquek Any suggestions to test the db stuffs up till this point?

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=9

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-9

File: src/context/db.tsx (hunk 1 above)
Comment (sebastianquek) [reply src/context/db.tsx]: Shall we add `"react/prop-types": 0` to our eslint? We used it here too: https://github.com/Open-Attestation/react-component-template/blob/master/package.json#L138

**Comment by sebastianquek** [comment]
> @sebastianquek Any suggestions to test the db stuffs up till this point?

Nope, none at the moment

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=9

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-9

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=9

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-9

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=9

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-9

**Pull request merged**